File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        if rowIndex==0:
            return [1]
        
        triangle=[[1]]

        for i in range(rowIndex):
            last_row=triangle[-1]
            new_row=[1]

            for j in range(1,len(last_row)):
                new_row.append(last_row[j-1]+last_row[j])
            
            new_row.append(1)
            triangle.append(new_row)
        
        return triangle[-1]

File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.cs
public class Solution {
    public IList<int> GetRow(int rowIndex) {
        if (rowIndex == 0) {
            return new int[] { 1 };
        }

        List<List<int>> triangle = new List<List<int>>();
        triangle.Add(new List<int> { 1 });

        for (int i = 0; i < rowIndex; i++) {
            List<int> lastRow = triangle[triangle.Count - 1];
            List<int> newRow = new List<int>();
            newRow.Add(1);

            for (int j = 1; j < lastRow.Count; j++) {
                newRow.Add(lastRow[j - 1] + lastRow[j]);
            }

            newRow.Add(1);
            triangle.Add(newRow);
        }

        return triangle[triangle.Count - 1].ToArray();
    }
    }